### DISCUSSION TITLE
Windows global utf8 encoding

### DISCUSSION THREAD

**Discussion by ZiyueWangUoB** [nlohmann/json#3847] [General]
Hi, my code with encoding utf8 in the json string works when I enable the global windows utf8 encoding, but breaks when it's disabled. Any way to get it working throughout?

**Comment by nlohmann** [comment]
No, the library only supports UTF-8.

**Comment by qmcreeper** [comment]
I add these code in CMakeLists.txt and it works.
```
add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/source-charset:utf-8>")
add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/execution-charset:utf-8>")
```
Maybe you can change code files encoding and runtime encoding in your compiler settings.
